src/chunk_constraint.c

```diff
@@ -853,13 +853,65 @@ chunk_constraint_rename_hypertable_from_tuple(TupleInfo *ti, const char *newname
 										   NameStr(new_chunk_constraint_name));
 
 	new_tuple = heap_modify_tuple(tuple, tupdesc, values, nulls, repl);
+
+	ts_chunk_index_adjust_meta(chunk_id,
+							   newname,
+							   NameStr(*old_chunk_constraint_name),
+							   NameStr(new_chunk_constraint_name));
+
 	ts_catalog_update(ti->scanrel, new_tuple);
 	heap_freetuple(new_tuple);
 
 	if (should_free)
 		heap_freetuple(tuple);
 }
 
+/*
+ * Adjust internal metadata after index/constraint rename
+ */
+int
+ts_chunk_constraint_adjust_meta(int32 chunk_id, const char *ht_constraint_name, const char *oldname,
```

Comment:
There's already a function `ts_chunk_constraint_rename_hypertable_constraint`. Why do we need also this function and what is the difference? It seems that there's a lot of code and functionality overlap since this function is doing mostly the same thing.

Reply:
The current code mixes DDL operations with metadata adjustments which doesnt work when multiple adjustments need to happen as those DDL operations are not idempotent. While it makes sense to refactor the code to separate DDL from metadata adjustment thats not the focus of this PR

Reply:
You can, for instance, just add a boolean parameter to the existing function  "apply_ddl=false` that doesn't do DDL changes when set to false and you can reuse the existing code.

Reply:
i think your suggesting changes actually decrease code quality, i initially had it all in a single function and decided against it. Also this is a bugfix PR that needs backporting into 1.7.x branch so it is beneficial to keep refactoring separate.